pkg/resourcecreator/deployment_test.go

```diff
@@ -332,6 +332,32 @@ func TestDeployment(t *testing.T) {
 		assert.NotEmpty(t, jwkerVolume)
 		assert.Equal(t, jwkerSecret, jwkerVolume.Name)
 		assert.Equal(t, jwkerSecret, jwkerVolume.VolumeSource.Secret.SecretName)
+
+		assert.Equal(t, 1, len(appContainer.EnvFrom))
+		assert.Equal(t, jwkerSecret, appContainer.EnvFrom[0].SecretRef.Name)
+	})
+
+	t.Run("jwker secrets are mounted as files only when MountSecretsAsFilesOnly is true", func(t *testing.T) {
```

Comment:
This description is ambiguous. It can be read in two ways:

1. Secrets are mounted as files if and only if MountSecretsAsFilesOnly is true. Implicitly, they are always available as env-variables.
2. Secrets are only mounted as files and not exposed as env-variables if MountSecretsAsFilesOnly is true.

I gather the second reading is the correct one, because the description of the flag is more explicit than this test description, but might be worth rewording this to be clearer.